# Задача для 4-го класса

## Формулировка

У Ивана и Петра вместе 980 рублей, у Ивана и Никиты вместе 930 рублей, а у Петра и Никиты вместе 890 рублей. Сколько денег у каждого из них?

## Импорт инструментов

In [1]:
import pandas as pd
%pylab inline

Populating the interactive namespace from numpy and matplotlib


## Исходные данные

In [2]:
# создаем имена для строк и столбцов
nms = ['Иван','Петр','Никита']
tog = ['Иван+Петр','Иван+Никита','Петр+Никита']

# задаем матрицу системы
a = pd.DataFrame([[1,1,0],[1,0,1],[0,1,1]], columns=nms, index=tog)

# задаем правую часть системы
b = pd.Series([980,930,890], index=tog)

## Печать исходных данных

In [3]:
print('---------------  матрица системы  ------------------')
print(a)
print('---------------  правая часть     ------------------')
print(b)

---------------  матрица системы  ------------------
             Иван  Петр  Никита
Иван+Петр       1     1       0
Иван+Никита     1     0       1
Петр+Никита     0     1       1
---------------  правая часть     ------------------
Иван+Петр      980
Иван+Никита    930
Петр+Никита    890
dtype: int64


## Решение системы уравнений

In [4]:
# решаем линейную систему уравнений
x = pd.Series(np.linalg.solve(a,b), index=nms).astype(int)

## Печать результата

In [5]:
print(x)

Иван      510
Петр      470
Никита    420
dtype: int32


## Интерпретация результата

У Ивана было 510 рублей, у Петра - 470 рублей, и у Никиты - 420 рублей.